backend/src/controller/chatController.ts

```diff
@@ -65,34 +122,31 @@ async function handleHigherLevelChat(
 	currentLevel: LEVEL_NAMES,
 	chatModel: ChatModel,
 	chatHistory: ChatHistoryMessage[],
-	defences: Defence[],
-	sentEmails: EmailInfo[]
+	defences: Defence[]
 ): Promise<LevelHandlerResponse> {
 	let updatedChatHistory = [...chatHistory];
-	let updatedChatResponse = {
+	const updatedChatResponse = {
```

Comment:
It's difficult to see where this is being mutated within this 80-line function. Wouldn't it be better to construct the new values to go into the updated chat response on-the-fly, and then combine them at the end into a new ChatHttpResponse?